src/CompositeRheologies.jl

```diff
@@ -206,13 +225,136 @@ function compute_εII_AD(v::Parallel, τII, args; tol=1e-6, verbose=false)
 end
 
+"""
+    p,τII = compute_p_τII(v::CompositeRheology, εII, εvol, args; tol=1e-6, verbose=false) 
+
+This updates pressure `p` and deviatoric stress invariant `τII` in case the composite rheology structure has volumetric components, but does not contain plastic or parallel elements.
+The 'old' pressure should be stored in `args` as `args.P_old`   
+"""
+function compute_p_τII(
+    v::CompositeRheology{T,N,
+                    Npar,is_parallel,
+                    Nplastic,is_plastic,
+                    Nvol,is_vol,
+                    false}, 
+        εII::_T, 
+        εvol::_T,
+        args; 
+        tol=1e-6, verbose=false
+    ) where {T, N, _T, Npar, is_parallel, Nplastic, is_plastic, Nvol, is_vol}
+
+    # A composite rheology case that may have volumetric elements, but the are not 
+    # tightly coupled, so we do NOT perform coupled iterations.
+    τII = local_iterations_εII(v, εII, args; tol=tol, verbose=verbose)
+    P   = local_iterations_εvol(v, εvol, args; tol=tol, verbose=verbose)
+
+    return P,τII
+end
+
+
+"""
+    p,τII = compute_p_τII(v::CompositeRheology, εII, εvol, args; tol=1e-6, verbose=false) 
+
+This updates pressure `p` and deviatoric stress invariant `τII` in case the composite rheology structure has no volumetric elemnts, but may contain plastic or parallel elements. 
+In that case, pressure is not updated (`args.P` is used instead).     
+"""
+function compute_p_τII(
+    v::CompositeRheology{T,N,
+                    Npar,is_parallel,
+                    Nplast,is_plastic,
+                    0,is_vol, false}, 
+        εII::_T, 
+        εvol::_T,
+        args;
+        tol=1e-6, verbose=false
+    ) where {T, N, _T, Npar, is_parallel, Nplast, is_plastic, Nvol, is_vol}
+    
+    # A composite rheology case with no parallel element; iterations for τII
+    τII, = local_iterations_εII(v, εII, args; tol=tol, verbose=verbose)
+    
+    P = any(keys(args) .=== :P_old) ? args.P_old : 0.0
+
+    return P, τII
+end
+
+"""
+    p,τII = compute_p_τII(v::CompositeRheology, εII, εvol, args; tol=1e-6, verbose=false) 
+
+This updates pressure `p` and deviatoric stress invariant `τII` in case the composite rheology structure has volumetric components and has volumetric plasticity
+The 'old' pressure should be stored in `args` as `args.P_old`   
+"""
+function compute_p_τII(
+    v::CompositeRheology{T,N,
+                    Npar,is_parallel,
+                    Nplastic,is_plastic,
+                    Nvol,is_vol,
+                    true}, 
+        εII::_T, 
+        εvol::_T,
+        args; 
+        tol=1e-6, verbose=false
+    ) where {T, N, _T, Npar, is_parallel, Nplastic, is_plastic, Nvol, is_vol}
+
+    # A composite rheology case that may have volumetric elements, but the are not 
+    # tightly coupled, so we do NOT perform coupled iterations.
+    out = local_iterations_εvol_εII(v, εII, εvol, args; tol=tol, verbose=verbose)
+
+    τII = out[1]
+    P = out[end]
+
+    return P,τII, out[2:N]
```

Comment:
is this necessary? Couldn't this function just return `out`?

Reply:
we expect  P,τII (in that order) in the following routine; in `out` it is ordered differently. we can ofcourse rearrange it in a follow-up routine. changing the order within out would also be possible somehow, but we would end up with having to rewrite parts of the routine (which would make this not reusable in the non-dilatant plasticity routine)

Reply:
As long as `out[2:N]` does not allocate, happy with it.

Reply:
doesn't:
```Julia
julia> εxx,εzz  = 7e-15, -6.8e-15
julia> εII  = sqrt(0.5*(εxx^2 + εzz^2))
julia> εvol = εxx + εzz
julia> args = (T = 900.0, d = 0.0001, τII_old = 700000.0, dt = 8.0e9, P = 0.0, P_old = 1e6)
julia> c16= CompositeRheology(SetConstantElasticity(G=1e10, Kb=2e11), LinearViscous(η=1e20), DruckerPrager_regularised(C=3e5, Ψ=10, η_vp=1e19))
--/\/\/----⟦▪̲̅▫̲̅▫̲̅▫̲̅----▬▬▬__--
julia> @btime compute_p_τII($c16, $εII, $εvol, $args)
  52.581 ns (0 allocations: 0 bytes)
(1.2583092027089968e6, 909776.9097816481, (2.081468729181813e-15, 1.2583092027089968e6))
```